spec/models/manageiq/providers/cloud_manager/vm_spec.rb

```diff
@@ -44,4 +48,158 @@
       end
     end
   end
+
+  context "queued methods" do
+    before do
+      allow(ems).to receive(:queue_name_for_ems_operations).and_return(queue_name)
+    end
+
+    it 'queues an associate floating IP task with associate_floating_ip_queue' do
+      ip_address = '1.2.3.4'
+      task_id = subject.associate_floating_ip_queue(user.userid, ip_address)
+
+      expect(MiqTask.find(task_id)).to have_attributes(
+        :name   => "associating floating IP with Instance for user #{user.userid}",
+        :state  => "Queued",
+        :status => "Ok"
+      )
+
+      expect(MiqQueue.where(:class_name => described_class.name).first).to have_attributes(
+        :class_name  => described_class.name,
+        :method_name => 'associate_floating_ip',
+        :role        => 'ems_operations',
+        :queue_name  => queue_name,
```

Comment:
Could you just do `:queue_name => ems.queue_name_for_ems_operations` here instead?

Reply:
I could, yes, I just thought it was better to be more explicit here in case the ems has something unexpected. Not that it would, but still. :)

Reply:
@agrare oops, actually, no, because the ems uses nil, but it gets set to "generic" internally.

Reply:
https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_queue.rb#L138

Reply:
Yeah I think we should change it from nil to explicit `"generic"`

Reply:
https://github.com/ManageIQ/manageiq/pull/19616#issuecomment-564169104

Reply:
@agrare ok https://github.com/ManageIQ/manageiq/pull/19627 :)